pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -235,11 +268,14 @@ func (configgen *ConfigGeneratorImpl) buildOutboundSniDnatClusters(env *model.En
 					}
 					subsetCluster := buildDefaultCluster(env, subsetClusterName, discoveryType, lbEndpoints, model.TrafficDirectionOutbound, proxy)
 					subsetCluster.TlsContext = nil
-					applyTrafficPolicy(env, subsetCluster, destinationRule.TrafficPolicy, port, nil, "",
-						SniDnatClusterMode, model.TrafficDirectionOutbound, proxy)
-					applyTrafficPolicy(env, subsetCluster, subset.TrafficPolicy, port, nil, "",
-						SniDnatClusterMode, model.TrafficDirectionOutbound, proxy)
+
+					opts.cluster = subsetCluster
```

Comment:
I think this makes it actually harder to follow what is going on, now I need to keep track of how opts was initialized, and find all the places fields changed to understand what is being passed. Before it was very explicit

Reply:
yeah, I can go either way. I still like extracting the variables into the buildClusterOpts object. i've modified the PR to keep things explicit.